mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -570,12 +570,11 @@ pub async fn init_p2p(ctx: MmArc) -> P2PResult<()> {
     let seednodes = seednodes(&ctx)?;
 
     let ctx_on_poll = ctx.clone();
-    let force_p2p_key = if i_am_seed {
+
+    let p2p_key = {
         let crypto_ctx = CryptoCtx::from_ctx(&ctx).mm_err(|e| P2PInitError::Internal(e.to_string()))?;
         let key = sha256(crypto_ctx.mm2_internal_privkey_slice());
-        Some(key.take())
-    } else {
-        None
```

Comment:
Also no login mode (`test_no_login` fails) probably due to this change

Reply:
Fixed.